### PR TITLE
Added support for co-author

### PR DESCRIPTION
**Pull request by CaptainDredge**
fixes #318 in grimoirelab-perceval

### CODE REVIEW

**Comment by Prabhat-IIT** [comment]
@acs @jgbarah I think failing of travis build has nothing to do with changes made by me. Am I right?

**Comment by jgbarah** [comment]
> fixes #318 in grimoirelab-perceval

I think you refer to chaoss/grimoirelab-perceval#318, right?

The patch looks good to me, @acs, could you please have a look at it?

> I think failing of travis build has nothing to do with changes made by me. Am I right?

Not completely sure, but yes, I think it has nothing to do with this patch. @acs do you agree?

**Review by jgbarah — APPROVED** [review]
LGTM

**Comment by Prabhat-IIT** [comment]
Moreover, @jgbarah  I'm still trying to get familiar with all the aspects of grimoirelab by  contributing to different issues. One, issue I face repeatedly is the testing. The tests won't run locally either due to configuration issues or some unmet dependencies which were no mentioned anywhere. I think some documentation related to various parts of code apart from general documentation of grimoirelab-tutorial is also need of the hour :)

**Comment by jgbarah** [comment]
I agree with the need of documentation. This said, for tests you can have a look a the `.travis` files, which in most cases will give you the needed clues.

**Comment by Prabhat-IIT** [comment]
Yup, at least I can enjoy the perk of `yaml` files being human readable :sweat_smile:

**Comment by acs** [comment]
> Not completely sure, but yes, I think it has nothing to do with this patch. @acs do you agree?

Yes, the tests are failing because a refactoring that has not arrived yet to the pip packages used in travis.

**Comment by acs** [comment]
Pretty interesting PR. Thanks @Prabhat-IIT . I find the code pretty clean. We must test it with more deep but until then, I approve it because it does not break anything. LGTM

File: grimoire_elk/enriched/git.py [hunk 1]
```diff
@@ -93,6 +93,7 @@ class GitEnrich(Enrich):
     # REGEX to extract authors from a multi author commit: several authors present
     # in the Author field in the commit. Used if self.pair_programming is True
     AUTHOR_P2P_REGEX = re.compile(r'(?P<first_authors>.* .*) and (?P<last_author>.* .*) (?P<email>.*)')
+    AUTHOR_P2P_NEW_REGEX = re.compile(r"Co-authored-by:(?P<first_authors>.* .*)<(?P<email>.*)>\n?")
```
Comment (acs) [inline grimoire_elk/enriched/git.py]: I would use AUTHOR_P2P_GITHUB_REGEX to make clearer that this feature is for adding support for co-authors in GitHub. And if it is not  only GitHub, something like AUTHOR_P2P_CO_AUTHOR_REGEX is better.

**Review by acs — APPROVED** [review]
LGTM

File: grimoire_elk/enriched/git.py (hunk 1 above)
Comment (Prabhat-IIT) [reply grimoire_elk/enriched/git.py]: @acs you have already merged it otherwise I'll make the suggested change in name :sweat_smile:

Comment (acs) [reply grimoire_elk/enriched/git.py]: you can create a new PR with the change :) But it is not so important. It is better to test it in some deep. I will try to find sometime to do it.

Comment (Prabhat-IIT) [reply grimoire_elk/enriched/git.py]: Sure, I think there can be changes in Regex for some extreme cases I couldn't have thought of!

**Pull request merged**